Tests/FluentAssertions.Specs/Types/MethodInfoSelectorSpecs.cs

```diff
@@ -272,6 +273,32 @@ public void When_selecting_methods_not_decorated_with_a_noninheritable_attribute
             methods.Should().ContainSingle();
         }
 
+        [Fact]
+        public void When_selecting_methods_that_are_async_it_should_only_return_the_applicable_methods()
+        {
+            // Arrange
+            Type type = typeof(TestClassForMethodSelectorWithAsync);
```

Comment:
As this class does not have any non-async methods, this test does not verify that `ThatAreAsync` excludes non-async methods.

How about adding a non-async method to the class and chain the `ContainSingle` with an assertion that the single method is the expected one?

You could then do almost the same in the test below to verify `ThatAreNotAsync`.

Reply:
I have changed the class accordingly.